ocs_ci/ocs/node.py

```diff
@@ -657,15 +657,17 @@ def get_compute_node_names(no_replace=False):
     """
     platform = config.ENV_DATA.get('platform').lower()
     compute_node_objs = get_typed_nodes()
-    if platform == constants.VSPHERE_PLATFORM or constants.AWS_PLATFORM:
+    if platform == (constants.VSPHERE_PLATFORM or constants.AWS_PLATFORM):
         return [
             compute_obj.get()['metadata']['labels'][constants.HOSTNAME_LABEL]
             for compute_obj in compute_node_objs
         ]
     elif (
-        platform == constants.BAREMETAL_PLATFORM
-        or platform == constants.BAREMETALPSI_PLATFORM
-        or platform == constants.IBM_POWER_PLATFORM
+        platform == (
+            constants.BAREMETAL_PLATFORM
```

Comment:
I believe verification passed because baremetal just happens to be the first in this list and that is what you tested on.

```python
In [1]: from ocs_ci.ocs import constants

In [2]: constants.BAREMETAL_PLATFORM or constants.BAREMETALPSI_PLATFORM or constants.IBM_POWER_PLATFORM
Out[2]: 'baremetal'

In [3]: platform = 'baremetal'

In [4]: platform == (constants.BAREMETAL_PLATFORM or constants.BAREMETALPSI_PLATFORM or constants.IBM_POWER_PLATFORM)
Out[4]: True
```

However if I change the order...
```python

In [5]: platform == (constants.BAREMETALPSI_PLATFORM or constants.IBM_POWER_PLATFORM or constants.BAREMETAL_PLATFORM)
Out[5]: False
```